plugins/module_utils/network/junos/facts/acls/acls.py

```diff
@@ -145,9 +145,17 @@ def render_config(self, spec, conf):
                     if term.get("from"):
                         if term["from"].get("source-address"):
                             ace["source"] = ace.get("source", {})
-                            ace["source"]["address"] = term["from"][
-                                "source-address"
-                            ]["name"]
+                            source_address = term["from"].get("source-address")
+                            if not isinstance(source_address, list):
```

Comment:
@rohitthakur2590 Maybe this is still worth double checking.

I was initially trying to see if I could get both forms work (support both str and list of str) hence the facts class has this alternate conditional which inserts the *string* to the dict if indeed the source is not a list. However, at least with a little bit of trial and error I couldn't get the Zuul tests to pass with both forms present. Thus ended up modifying all test cases to be a list, aware that this could be a breaking change for users.

If indeed modifying the argspec is a realistic option, I'm still wondering should we

1) remove this extra conditional
or
2) modify it so that it constructs a list if the source data is not a list (similar to the conditional in config/acls/acls.py)

?

Reply:
@junousi  Yes, I think you have a very good point here. Now that I was giving it a thought, Instead of making it a breaking change, we can actually support both types str/list for address. 
and to have this sort of solution we can have address as raw input

Reply:
I believe most comments are now addressed, and also tests reverted back so that the single-address test cases remain as they were.

The only open question for me is how should this facter part behave when encountering non-list input.

Reply:
So in that case that actually behaves the old way i.e having address as a string.